### PR TITLE
Develop

### LINKED ISSUE #18: Integrate Sprint 3 Features and Fixes into Develop Branch kotlin
This issue involves merging all completed features and fixes from Sprint 3 into the develop branch. The integration is crucial to validate that all individual components work together without conflicts and that the system as a whole operates smoothly. This step is necessary before merging into the main branch.

Tasks:

- Ensure all Sprint 3 features and fixes are finalized and tested individually.
- Begin the merging process into the develop branch.
- Resolve any merge conflicts that arise during the integration.
- Conduct comprehensive testing to ensure that all components work together as expected.
- Document any bugs or issues discovered during testing.
- Fix identified issues and perform additional testing as needed to achieve stability.

Acceptance Criteria:

- All features and fixes from Sprint 3 are merged into the develop branch.
- The develop branch passes all integration tests, with all parts functioning cohesively.
- No high-priority issues remain unresolved in the develop branch.

Priority: High

Related Milestone: Integration of Sprint 3 into Develop Branch

### PR DESCRIPTION
**Pull request by lojedaa**
Re-implement notification functionality for items in cart. 

- Restored notification system that alerts users when items are in the cart, which was accidentally removed in a previous commit.
- Verified that notifications are correctly triggered both when the cart has items and when it's empty.
- Tested all related functionalities to ensure no regressions occurred.
- Ensured compatibility with other features and improved test coverage for edge cases.

### CODE REVIEW

File: app/src/main/java/com/example/ecostyle/view/CheckoutFragment.kt [hunk 1]
```diff
@@ -42,28 +48,30 @@ class CheckoutFragment : Fragment() {
...
-
-        // Verificar stock y proceder al pago
+        // Comportamiento del botón de checkout
         checkoutButton.setOnClickListener {
-            verifyStockBeforeCheckout()
+            if (isNetworkAvailable(requireContext())) {
+                verifyStockBeforeCheckout()
+            } else {
+                Toast.makeText(context, "\n" +
+                        "Payment cannot be made offline. Please try again later.", Toast.LENGTH_LONG).show()
```
Comment (Feru34) [inline app/src/main/java/com/example/ecostyle/view/CheckoutFragment.kt]:
```suggestion
                        "Payment cannot be made offline. Please try again later.", Toast.LENGTH_SHORT).show()
```
creo que deberiamos tener los toast cortos, a veces se sobrelapan

**Review by Feru34 — CHANGES_REQUESTED** [review]

**Review by Feru34 — APPROVED** [review]
